manageiq-operator/api/v1alpha1/helpers/miq-components/util.go

```diff
@@ -144,6 +145,22 @@ func addOrUpdateEnvVar(environment []corev1.EnvVar, variable corev1.EnvVar) []co
 	return environment
 }
 
+func addOrUpdateProjectedSecretVolumeSource(volumeName string, volumes []corev1.Volume, volumeProjection *corev1.VolumeProjection) corev1.ProjectedVolumeSource {
+	projectedVolumeSource := corev1.ProjectedVolumeSource{}
+
+	if volume := miqutilsv1alpha1.FindVolume(volumeName, volumes); volume.VolumeSource.Projected != nil {
+		if foundVolumeProjection := miqutilsv1alpha1.FindVolumeProjection((*volumeProjection).Secret.LocalObjectReference.Name, volume.VolumeSource.Projected.Sources); foundVolumeProjection.Secret != nil {
+			projectedVolumeSource.Sources = volume.VolumeSource.Projected.Sources
+		} else {
+			projectedVolumeSource.Sources = append(volume.VolumeSource.Projected.Sources, *volumeProjection)
+		}
+	} else {
+		projectedVolumeSource.Sources = []corev1.VolumeProjection{*volumeProjection}
+	}
+
+	return projectedVolumeSource
+}
```

Comment:
Not sure if there was a better way to do this but this is essentially covering three scenarios:

1. If the projection already exists in the list of projected volume sources, then we return the existing list of projected volume sources
2. If the projection doesn't exist in the list of projected volume sources, then we append the projection to the source list
3. If the projected volume source list is empty, then we simple add the project to the list